Updated project, source, and tests to Swift 3.0

regynald opened the pull request:
Updated cartfile to use 3.0 versions for both Quick & Nimble.
Had to remove typealias Middleware since Swift 3.0 doesn't allow @escaping in type aliases

lsunsi committed:
Merge branch 'release/v1.0.0'

Commit:
Updated project, source, and tests to Swift 3.0

Updated cartfile to use 3.0 versions for both Quick & Nimble.
Had to remove typealias Middleware since Swift 3.0 doesn't allow @escaping in type aliases

regynald commented:
Escaping change is [here](https://github.com/apple/swift/pull/4388/files)

lsunsi commented on Cartfile.private:
```diff
@@ -1,3 +1,2 @@
-github "Quick/Nimble"
-
+github "Quick/Quick" "swift-3.0"
+github "Quick/Nimble" "master"
```
this is fine for develop, but for a official master release we should probably wait until proper swift 3 support is released on both os this packages

lsunsi commented on ReduxSwift.playground/Contents.swift:
```diff
@@ -7,7 +7,7 @@ struct AppState {
         var done: Bool
     }
-
+    
```
trailing whitespace. please use swiftlint (config file is included in the project) so we can maintain a coherent code style in the project. ps: this also applied for other styling changes

lsunsi commented on Sources/Core.swift:
```diff
@@ -8,18 +8,22 @@
+            self.state = reducer(action, self.state)
+            return action
+        }) {$1(self, $0)}*/
+        
```
I don't get this comment. Seems to me it could just be removed.

lsunsi commented on Sources/Subscriptions.swift:
```diff
@@ -6,7 +6,7 @@
 //
 
-public class Subscriptions<State, Key: Hashable> {
+open class Subscriptions<State, Key: Hashable> {
```
maybe this class should be public and not open. what do you think?
my reasoning about this is that this class (and member) aren't intended for subclassing

regynald replied:
Yeah, I'm just gonna make the subscription class and associated functions public.

lsunsi reviewed (changes requested):
General comments on code style and some questions. Also, please merge on develop and not on master, so I can wait on Quick/Nimble next releases and properly tag ReduxSwift releases

Commit:
Public classes for subscriptions and dependencies update.

Changed the subscription class and associated functions to be public to
disallow subclasses. Using master branches for both quick & nimble.
